File: Julia/gan-mode-collapse.jl
function detect_mode_collapse(generated_samples; threshold::Float64=0.1)
    feature_stds = mapslices(std, generated_samples; dims=1)
    diversity_score = mean(feature_stds)
    is_collapsed = diversity_score < threshold
    return (diversity_score = diversity_score, is_collapsed = is_collapsed)
end

File: R/gan-mode-collapse.R
detect_mode_collapse <- function(generated_samples, threshold = 0.1) {
  feature_stds <- apply(generated_samples, 2, sd)
  diversity_score <- mean(feature_stds)
  is_collapsed <- diversity_score < threshold
  list(diversity_score = diversity_score, is_collapsed = is_collapsed)
}
